### PR TITLE
feature: Add check your answers page with all functionality

### PR DESCRIPTION
**Pull request by Zain-Abbas1**
- Make page dynamic displaying data form db
- Allow data to be saved to backend
- Create new unit tests

### CODE REVIEW

File: server/routes/report.js [hunk 1]
```diff
@@ -364,12 +364,67 @@ router.post('/report-a-discrepancy/discrepancy-details', (req, res, next) => {
...
+      viewData.this_data.companyNumber = report.data.company_number.toUpperCase();
+      return api.companyProfile.getCompanyProfile(report.data.company_number.toUpperCase());
+    }).then(profile => {
+      viewData.this_data.companyName = profile.resource.companyName;
+      viewData.this_data.companyNumber = profile.resource.companyNumber;
+    }).then(_ => {
+      const session = res.locals.session;
+      viewData.this_data.pscName = session.appData.selectedPscDetails.name;
+      // Discrepancy details still to be done
+      viewData.this_data.additionalDetails = session.appData.selectedPscDetails.details;
```
Comment (CH-JonathanBlake) [inline server/routes/report.js:388]: Why does this need to be in a different then()?

File: server/routes/report.js [hunk 2]
```diff
@@ -364,12 +364,67 @@ router.post('/report-a-discrepancy/discrepancy-details', (req, res, next) => {
...
+      data.obliged_entity_type = report.data.obliged_entity_type;
+      data.obliged_entity_organisation_name = report.data.obliged_entity_organisation_name;
+      data.obliged_entity_contact_name = report.data.obliged_entity_contact_name;
+      data.obliged_entity_email = report.data.obliged_entity_email;
+      data.company_number = report.data.company_number;
+      data.etag = report.data.etag;
+      return pscDiscrepancyService.saveStatus(data);
+    }).then(_ => {
+      res.redirect(302, '/report-a-discrepancy/confirmation');
+    }).catch(err => {
```
Comment (CH-JonathanBlake) [inline server/routes/report.js:419]: Do we actually have to wait for the response from the save if we're not continuing?
If we are we may as well check for a successful response code.

Comment (ICraig-CH) [reply server/routes/report.js:419]: I think the errors are caught as long as its chained, ive had errors happen in a post and it goes to the error page

**Review by ICraig-CH — APPROVED** [review]

File: server/routes/report.js (hunk 1 above)
Comment (ch-rfishwick) [reply server/routes/report.js:388]: I think I'm happy to approve on the basis that I can't see a problem being caused here, but would be nice to know 👍

**Review by ch-rfishwick — APPROVED** [review]

File: server/routes/report.js (hunk 2 above)
Comment (Zain-Abbas1) [reply server/routes/report.js:419]: I think what Iain is saying makes sense, and seems to be similar to what I've seen with this.

File: server/routes/report.js (hunk 1 above)
Comment (Zain-Abbas1) [reply server/routes/report.js:388]: Correct, apart from a readability aspect this can probably be combine with the previous then. I'll change now

Comment (Zain-Abbas1) [reply server/routes/report.js:387]: Done

**Review by CH-JonathanBlake — APPROVED** [review]

**Pull request merged**